funding/manager.go

```diff
@@ -2182,16 +2182,37 @@ func (f *Manager) continueFundingAccept(resCtx *reservationWithCtx,
 	fundingCreated.CommitSig, err = lnwire.NewSigFromSignature(sig)
 	if err != nil {
 		log.Errorf("Unable to parse signature: %v", err)
-		f.failFundingFlow(resCtx.peer, pendingChanID, err)
+		f.failFundingFlowAndRemoveChannel(
+			resCtx.peer, channelID, pendingChanID, err,
+		)
 		return
 	}
 	if err := resCtx.peer.SendMessage(true, fundingCreated); err != nil {
 		log.Errorf("Unable to send funding complete message: %v", err)
-		f.failFundingFlow(resCtx.peer, pendingChanID, err)
+		f.failFundingFlowAndRemoveChannel(
+			resCtx.peer, channelID, pendingChanID, err,
+		)
 		return
 	}
 }
 
+// failFundingFlowAndRemoveChannel removes the pending channel from Brontide
+// and fails the funding flow.
+func (f *Manager) failFundingFlowAndRemoveChannel(peer lnpeer.Peer,
```

Comment:
I think there are other cases where we also need to remove the pending channel.  For example, the `fundingTimeout` case.

WDYT about putting the `RemovePendingChannel` directly into `failFundingFlow` and then checking for the pending channel on the brontide side (and doing nothing if it doesn't exist)?

Reply:
Good catch on the `fundingTimeout` case! Do you think there are more cases? 

> WDYT about putting the RemovePendingChannel directly into failFundingFlow and then checking for the pending channel on the brontide side (and doing nothing if it doesn't exist)?

That's actually my initial approach. It's ditched because I wanna finer control over the communication between the funding manager and the brontide. Plus there are way more `failFundingFlow` calls that use a temp ID, and the current approach seems like a more minor change.

Reply:
Any failure in `handleFundingSigned`.  I think also some failures in `pruneZombieReservations` may happen when a channel is pending.

Reply:
Basically any calls to `failFundingFlow` after `AddPendingChannel`. We have to be careful with `handleFundingSigned` though and validate that the `msg.ChanID` is actually _from_ the peer in question. This might also be a good time to improve `signedReservations`. There's not an issue there afaict, but the design is wrong and the map should also be keyed by peer id